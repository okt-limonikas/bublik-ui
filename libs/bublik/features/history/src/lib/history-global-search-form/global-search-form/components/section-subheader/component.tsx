/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { FC } from 'react';

export interface FormSectionSubheaderProps {
	name: string;
}

export const FormSectionSubheader: FC<FormSectionSubheaderProps> = ({
	name
}) => {
	return (
		<div className="mb-2">
			<span className="text-[0.6875rem] leading-[0.875rem]">{name}</span>
		</div>
	);
};
