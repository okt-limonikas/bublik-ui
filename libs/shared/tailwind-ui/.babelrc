{
	"presets": [
		[
			"@nx/react/babel",
			{
				"runtime": "automatic",
				"useBuiltIns": "usage"
			}
		],
		"@emotion/babel-preset-css-prop"
	],
	"plugins": []
}
