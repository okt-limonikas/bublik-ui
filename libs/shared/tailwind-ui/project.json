{
	"name": "shared-tailwind-ui",
	"$schema": "../../../node_modules/nx/schemas/project-schema.json",
	"sourceRoot": "libs/shared/tailwind-ui/src",
	"projectType": "library",
	"tags": [],
	"targets": {
		"lint": {
			"executor": "@nx/eslint:lint"
		},
		"test": {
			"executor": "@nx/vite:test",
			"outputs": ["{options.reportsDirectory}"],
			"options": {
				"reportsDirectory": "../../../coverage/libs/shared/tailwind-ui"
			}
		}
	}
}
