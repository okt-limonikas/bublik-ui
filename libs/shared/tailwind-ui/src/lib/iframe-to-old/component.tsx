/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, FC } from 'react';

import { cn } from '../utils';

export type IframeToOldBublikProps = ComponentPropsWithRef<'iframe'>;

export const IframeToOldBublik: FC<IframeToOldBublikProps> = (props) => {
	return (
		<iframe
			className={cn('w-full h-full border-none', props.className)}
			title="Old Bublik"
			{...props}
		/>
	);
};
