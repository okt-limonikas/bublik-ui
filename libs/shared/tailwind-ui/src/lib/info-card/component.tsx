/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, forwardRef } from 'react';

export interface InfoCardProps extends ComponentPropsWithRef<'div'> {
	header: string;
	message: string;
}

export const InfoCard = forwardRef<HTMLDivElement, InfoCardProps>(
	(props, ref) => {
		const { header, message } = props;

		return (
			<div
				className="flex items-center justify-center w-full h-screen"
				data-testid="info-card"
			>
				<div
					className="bg-white min-w-[600px] max-w-[768px] flex flex-col rounded-2xl shadow-2xl"
					ref={ref}
				>
					<div className="p-4 border-b border-border-primary">
						<h2>{header}</h2>
					</div>
					<div className="flex-1 p-4">
						<p>{message}</p>
					</div>
				</div>
			</div>
		);
	}
);
