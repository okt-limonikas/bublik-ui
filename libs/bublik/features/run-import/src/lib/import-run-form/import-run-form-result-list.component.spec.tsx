/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { Dialog, DialogContent } from '@/shared/tailwind-ui';
import { render } from '@testing-library/react';
import { QueryParamProvider } from 'use-query-params';
import { WindowHistoryAdapter } from 'use-query-params/adapters/window';
import { vi } from 'vitest';

import { RunImportResult } from './import-run-form-result-list.component';
import { ImportLogProvider } from '../import-events-table/import-log.component';

describe('ImportRunFormResultList', () => {
	beforeEach(() => {
		// Mock the date to a fixed timestamp in UTC
		vi.useFakeTimers();
		vi.setSystemTime(new Date('2024-01-01T12:00:00Z'));
	});

	afterEach(() => {
		vi.useRealTimers();
	});

	it('should render successfully', () => {
		const { asFragment } = render(
			<QueryParamProvider adapter={WindowHistoryAdapter}>
				<ImportLogProvider>
					<Dialog open>
						<DialogContent>
							<RunImportResult
								results={[
									{
										url: 'https://www.youtube.com/watch?v=1',
										taskId: '1'
									},
									{
										url: 'https://www.youtube.com/watch?v=2',
										taskId: '2'
									},
									{
										url: 'https://www.youtube.com/watch?v=3',
										taskId: '3'
									}
								]}
							/>
						</DialogContent>
					</Dialog>
				</ImportLogProvider>
			</QueryParamProvider>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
