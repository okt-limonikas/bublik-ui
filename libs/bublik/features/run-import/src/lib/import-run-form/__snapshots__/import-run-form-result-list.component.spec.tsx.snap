// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImportRunFormResultList > should render successfully 1`] = `
<DocumentFragment>
  <div
    aria-describedby="radix-:r5:"
    aria-labelledby="radix-:r4:"
    data-state="open"
    id="radix-:r3:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <div>
      <h2
        class="text-lg font-semibold leading-none tracking-tight"
        id="radix-:r4:"
      >
        Import Runs
      </h2>
      <p
        class="mt-1.5 text-base text-gray-500"
        id="radix-:r5:"
      >
        Scheduled runs will be imported in the background
      </p>
      <p
        class="mt-1.5 mb-6 text-sm text-gray-500"
      >
        You can check the logs and flower tasks
      </p>
      <h3
        class="text-sm font-medium text-text-primary mb-2"
      >
        Imports
      </h3>
      <ul
        class="border rounded-md [&>*:not(:last-child)]:border-b [&>*]:border-border-primary"
      >
        <li
          class="space-y-2 p-4"
        >
          <div
            class="flex items-center justify-between gap-2"
          >
            <span
              class="inline-flex px-2 py-0.5 text-xs font-semibold leading-5 rounded items-center bg-badge-3 text-text-expected"
            >
              <svg
                class="h-4 w-4 text-text-expected mr-1"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.99992 1.33301C4.32392 1.33301 1.33325 4.32301 1.33325 7.99967C1.33325 8.33967 1.35925 8.68034 1.40992 9.01234C1.44992 9.27701 1.69592 9.46167 1.96192 9.41768C2.22659 9.37701 2.40792 9.12967 2.36792 8.86567C2.32459 8.58167 2.30259 8.29167 2.30259 7.99967C2.30259 4.85834 4.85859 2.30234 7.99992 2.30234C11.1413 2.30234 13.6973 4.85834 13.6973 7.99967C13.6973 11.1417 11.1413 13.697 7.99992 13.697C6.21792 13.697 4.57192 12.8863 3.48325 11.473C3.31925 11.2617 3.01525 11.221 2.80392 11.385C2.59125 11.5477 2.55192 11.8523 2.71525 12.065C3.98859 13.7177 5.91525 14.6663 7.99992 14.6663C11.6759 14.6663 14.6666 11.6757 14.6666 7.99967C14.6666 4.32301 11.6759 1.33301 7.99992 1.33301Z"
                  fill="currentColor"
                />
                <path
                  d="M10.2265 6.06447L7.41587 8.8758L6.18721 7.64647C5.99187 7.45114 5.67521 7.45114 5.47987 7.64647C5.28454 7.8418 5.28454 8.15847 5.47987 8.3538L7.06187 9.93647C7.15587 10.0298 7.28254 10.0825 7.41587 10.0825C7.54787 10.0825 7.67521 10.0298 7.76921 9.93647L10.9339 6.7718C11.1292 6.57647 11.1292 6.2598 10.9339 6.06447C10.7385 5.86914 10.4219 5.86914 10.2265 6.06447Z"
                  fill="currentColor"
                />
              </svg>
              STARTED
            </span>
            <span
              class="text-xs text-gray-500"
            >
              03:00 PM
            </span>
          </div>
          <a
            class="rounded-md bg-primary-wash p-2 text-sm font-mono block hover:underline"
            href="https://www.youtube.com/watch?v=1"
            rel="noreferrer"
            target="_blank"
          >
            https://www.youtube.com/watch?v=1
          </a>
          <div
            class="flex gap-2"
          >
            <button
              class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none border border-border-primary text-text-primary hover:bg-gray-50 disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 py-[7px] text-[0.875rem] leading-[1.5rem] rounded-md flex-1"
            >
              <svg
                class="mr-1.5"
                fill="none"
                height="20"
                viewBox="0 0 28 28"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m23.674 8.816-5.891-6.198a.867.867 0 0 0-.63-.27h-.24c-.023-.003-.042-.015-.066-.015s-.044.012-.067.014H9.415c-2.89 0-5.331 2.41-5.331 5.261v4.852c0 .487.39.881.872.881a.877.877 0 0 0 .872-.88V7.607c0-1.897 1.642-3.499 3.587-3.499h6.56v2.523c0 2.15 1.73 3.9 3.858 3.906h.001a.877.877 0 0 0 .873-.879.876.876 0 0 0-.87-.883c-1.168-.003-2.117-.965-2.117-2.144V5.097l4.452 4.683v10.413c0 1.977-1.619 3.711-3.465 3.711H9.415c-2.01 0-3.587-1.63-3.587-3.711v-2.944a.877.877 0 0 0-.872-.88.877.877 0 0 0-.873.88v2.944c0 3.069 2.343 5.473 5.332 5.473h9.292c2.824 0 5.21-2.506 5.21-5.473V9.426a.887.887 0 0 0-.243-.61Z"
                  fill="currentColor"
                />
                <path
                  d="M10.307 13.492h7.395a.74.74 0 0 0 .733-.746.74.74 0 0 0-.733-.746h-7.404a.74.74 0 0 0-.733.746.74.74 0 0 0 .733.746h.01ZM10.046 17.345h4.796c.263 0 .476-.327.476-.73 0-.403-.213-.73-.476-.73H10.04c-.262 0-.475.327-.475.73 0 .403.213.73.476.73h.005Z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Log
              </span>
            </button>
            <a
              class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none border border-border-primary text-text-primary hover:bg-gray-50 disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 py-[7px] text-[0.875rem] leading-[1.5rem] rounded-md flex-1"
              href="http://localhost:3000/flower/task/1"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="size-4 mr-1.5"
                fill="none"
                height="15"
                viewBox="0 0 15 15"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.85357 3.85355L7.65355 3.05353C8.2981 2.40901 9.42858 1.96172 10.552 1.80125C11.1056 1.72217 11.6291 1.71725 12.0564 1.78124C12.4987 1.84748 12.7698 1.97696 12.8965 2.10357C13.0231 2.23018 13.1526 2.50125 13.2188 2.94357C13.2828 3.37086 13.2779 3.89439 13.1988 4.44801C13.0383 5.57139 12.591 6.70188 11.9464 7.34645L7.49999 11.7929L6.35354 10.6465C6.15827 10.4512 5.84169 10.4512 5.64643 10.6465C5.45117 10.8417 5.45117 11.1583 5.64643 11.3536L7.14644 12.8536C7.34171 13.0488 7.65829 13.0488 7.85355 12.8536L8.40073 12.3064L9.57124 14.2572C9.65046 14.3893 9.78608 14.4774 9.9389 14.4963C10.0917 14.5151 10.2447 14.4624 10.3535 14.3536L12.3535 12.3536C12.4648 12.2423 12.5172 12.0851 12.495 11.9293L12.0303 8.67679L12.6536 8.05355C13.509 7.19808 14.0117 5.82855 14.1887 4.58943C14.2784 3.9618 14.2891 3.33847 14.2078 2.79546C14.1287 2.26748 13.9519 1.74482 13.6035 1.39645C13.2552 1.04809 12.7325 0.871332 12.2045 0.792264C11.6615 0.710945 11.0382 0.721644 10.4105 0.8113C9.17143 0.988306 7.80189 1.491 6.94644 2.34642L6.32322 2.96968L3.07071 2.50504C2.91492 2.48278 2.75773 2.53517 2.64645 2.64646L0.646451 4.64645C0.537579 4.75533 0.484938 4.90829 0.50375 5.0611C0.522563 5.21391 0.61073 5.34954 0.742757 5.42876L2.69364 6.59928L2.14646 7.14645C2.0527 7.24022 2.00002 7.3674 2.00002 7.50001C2.00002 7.63261 2.0527 7.75979 2.14646 7.85356L3.64647 9.35356C3.84173 9.54883 4.15831 9.54883 4.35357 9.35356C4.54884 9.1583 4.54884 8.84172 4.35357 8.64646L3.20712 7.50001L3.85357 6.85356L6.85357 3.85355ZM10.0993 13.1936L9.12959 11.5775L11.1464 9.56067L11.4697 11.8232L10.0993 13.1936ZM3.42251 5.87041L5.43935 3.85356L3.17678 3.53034L1.80638 4.90074L3.42251 5.87041ZM2.35356 10.3535C2.54882 10.1583 2.54882 9.8417 2.35356 9.64644C2.1583 9.45118 1.84171 9.45118 1.64645 9.64644L0.646451 10.6464C0.451188 10.8417 0.451188 11.1583 0.646451 11.3535C0.841713 11.5488 1.1583 11.5488 1.35356 11.3535L2.35356 10.3535ZM3.85358 11.8536C4.04884 11.6583 4.04885 11.3417 3.85359 11.1465C3.65833 10.9512 3.34175 10.9512 3.14648 11.1465L1.14645 13.1464C0.95119 13.3417 0.951187 13.6583 1.14645 13.8535C1.34171 14.0488 1.65829 14.0488 1.85355 13.8536L3.85358 11.8536ZM5.35356 13.3535C5.54882 13.1583 5.54882 12.8417 5.35356 12.6464C5.1583 12.4512 4.84171 12.4512 4.64645 12.6464L3.64645 13.6464C3.45119 13.8417 3.45119 14.1583 3.64645 14.3535C3.84171 14.5488 4.1583 14.5488 4.35356 14.3535L5.35356 13.3535ZM9.49997 6.74881C10.1897 6.74881 10.7488 6.1897 10.7488 5.5C10.7488 4.8103 10.1897 4.25118 9.49997 4.25118C8.81026 4.25118 8.25115 4.8103 8.25115 5.5C8.25115 6.1897 8.81026 6.74881 9.49997 6.74881Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Task
              </span>
            </a>
          </div>
        </li>
        <li
          class="space-y-2 p-4"
        >
          <div
            class="flex items-center justify-between gap-2"
          >
            <span
              class="inline-flex px-2 py-0.5 text-xs font-semibold leading-5 rounded items-center bg-badge-3 text-text-expected"
            >
              <svg
                class="h-4 w-4 text-text-expected mr-1"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.99992 1.33301C4.32392 1.33301 1.33325 4.32301 1.33325 7.99967C1.33325 8.33967 1.35925 8.68034 1.40992 9.01234C1.44992 9.27701 1.69592 9.46167 1.96192 9.41768C2.22659 9.37701 2.40792 9.12967 2.36792 8.86567C2.32459 8.58167 2.30259 8.29167 2.30259 7.99967C2.30259 4.85834 4.85859 2.30234 7.99992 2.30234C11.1413 2.30234 13.6973 4.85834 13.6973 7.99967C13.6973 11.1417 11.1413 13.697 7.99992 13.697C6.21792 13.697 4.57192 12.8863 3.48325 11.473C3.31925 11.2617 3.01525 11.221 2.80392 11.385C2.59125 11.5477 2.55192 11.8523 2.71525 12.065C3.98859 13.7177 5.91525 14.6663 7.99992 14.6663C11.6759 14.6663 14.6666 11.6757 14.6666 7.99967C14.6666 4.32301 11.6759 1.33301 7.99992 1.33301Z"
                  fill="currentColor"
                />
                <path
                  d="M10.2265 6.06447L7.41587 8.8758L6.18721 7.64647C5.99187 7.45114 5.67521 7.45114 5.47987 7.64647C5.28454 7.8418 5.28454 8.15847 5.47987 8.3538L7.06187 9.93647C7.15587 10.0298 7.28254 10.0825 7.41587 10.0825C7.54787 10.0825 7.67521 10.0298 7.76921 9.93647L10.9339 6.7718C11.1292 6.57647 11.1292 6.2598 10.9339 6.06447C10.7385 5.86914 10.4219 5.86914 10.2265 6.06447Z"
                  fill="currentColor"
                />
              </svg>
              STARTED
            </span>
            <span
              class="text-xs text-gray-500"
            >
              03:00 PM
            </span>
          </div>
          <a
            class="rounded-md bg-primary-wash p-2 text-sm font-mono block hover:underline"
            href="https://www.youtube.com/watch?v=2"
            rel="noreferrer"
            target="_blank"
          >
            https://www.youtube.com/watch?v=2
          </a>
          <div
            class="flex gap-2"
          >
            <button
              class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none border border-border-primary text-text-primary hover:bg-gray-50 disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 py-[7px] text-[0.875rem] leading-[1.5rem] rounded-md flex-1"
            >
              <svg
                class="mr-1.5"
                fill="none"
                height="20"
                viewBox="0 0 28 28"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m23.674 8.816-5.891-6.198a.867.867 0 0 0-.63-.27h-.24c-.023-.003-.042-.015-.066-.015s-.044.012-.067.014H9.415c-2.89 0-5.331 2.41-5.331 5.261v4.852c0 .487.39.881.872.881a.877.877 0 0 0 .872-.88V7.607c0-1.897 1.642-3.499 3.587-3.499h6.56v2.523c0 2.15 1.73 3.9 3.858 3.906h.001a.877.877 0 0 0 .873-.879.876.876 0 0 0-.87-.883c-1.168-.003-2.117-.965-2.117-2.144V5.097l4.452 4.683v10.413c0 1.977-1.619 3.711-3.465 3.711H9.415c-2.01 0-3.587-1.63-3.587-3.711v-2.944a.877.877 0 0 0-.872-.88.877.877 0 0 0-.873.88v2.944c0 3.069 2.343 5.473 5.332 5.473h9.292c2.824 0 5.21-2.506 5.21-5.473V9.426a.887.887 0 0 0-.243-.61Z"
                  fill="currentColor"
                />
                <path
                  d="M10.307 13.492h7.395a.74.74 0 0 0 .733-.746.74.74 0 0 0-.733-.746h-7.404a.74.74 0 0 0-.733.746.74.74 0 0 0 .733.746h.01ZM10.046 17.345h4.796c.263 0 .476-.327.476-.73 0-.403-.213-.73-.476-.73H10.04c-.262 0-.475.327-.475.73 0 .403.213.73.476.73h.005Z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Log
              </span>
            </button>
            <a
              class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none border border-border-primary text-text-primary hover:bg-gray-50 disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 py-[7px] text-[0.875rem] leading-[1.5rem] rounded-md flex-1"
              href="http://localhost:3000/flower/task/2"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="size-4 mr-1.5"
                fill="none"
                height="15"
                viewBox="0 0 15 15"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.85357 3.85355L7.65355 3.05353C8.2981 2.40901 9.42858 1.96172 10.552 1.80125C11.1056 1.72217 11.6291 1.71725 12.0564 1.78124C12.4987 1.84748 12.7698 1.97696 12.8965 2.10357C13.0231 2.23018 13.1526 2.50125 13.2188 2.94357C13.2828 3.37086 13.2779 3.89439 13.1988 4.44801C13.0383 5.57139 12.591 6.70188 11.9464 7.34645L7.49999 11.7929L6.35354 10.6465C6.15827 10.4512 5.84169 10.4512 5.64643 10.6465C5.45117 10.8417 5.45117 11.1583 5.64643 11.3536L7.14644 12.8536C7.34171 13.0488 7.65829 13.0488 7.85355 12.8536L8.40073 12.3064L9.57124 14.2572C9.65046 14.3893 9.78608 14.4774 9.9389 14.4963C10.0917 14.5151 10.2447 14.4624 10.3535 14.3536L12.3535 12.3536C12.4648 12.2423 12.5172 12.0851 12.495 11.9293L12.0303 8.67679L12.6536 8.05355C13.509 7.19808 14.0117 5.82855 14.1887 4.58943C14.2784 3.9618 14.2891 3.33847 14.2078 2.79546C14.1287 2.26748 13.9519 1.74482 13.6035 1.39645C13.2552 1.04809 12.7325 0.871332 12.2045 0.792264C11.6615 0.710945 11.0382 0.721644 10.4105 0.8113C9.17143 0.988306 7.80189 1.491 6.94644 2.34642L6.32322 2.96968L3.07071 2.50504C2.91492 2.48278 2.75773 2.53517 2.64645 2.64646L0.646451 4.64645C0.537579 4.75533 0.484938 4.90829 0.50375 5.0611C0.522563 5.21391 0.61073 5.34954 0.742757 5.42876L2.69364 6.59928L2.14646 7.14645C2.0527 7.24022 2.00002 7.3674 2.00002 7.50001C2.00002 7.63261 2.0527 7.75979 2.14646 7.85356L3.64647 9.35356C3.84173 9.54883 4.15831 9.54883 4.35357 9.35356C4.54884 9.1583 4.54884 8.84172 4.35357 8.64646L3.20712 7.50001L3.85357 6.85356L6.85357 3.85355ZM10.0993 13.1936L9.12959 11.5775L11.1464 9.56067L11.4697 11.8232L10.0993 13.1936ZM3.42251 5.87041L5.43935 3.85356L3.17678 3.53034L1.80638 4.90074L3.42251 5.87041ZM2.35356 10.3535C2.54882 10.1583 2.54882 9.8417 2.35356 9.64644C2.1583 9.45118 1.84171 9.45118 1.64645 9.64644L0.646451 10.6464C0.451188 10.8417 0.451188 11.1583 0.646451 11.3535C0.841713 11.5488 1.1583 11.5488 1.35356 11.3535L2.35356 10.3535ZM3.85358 11.8536C4.04884 11.6583 4.04885 11.3417 3.85359 11.1465C3.65833 10.9512 3.34175 10.9512 3.14648 11.1465L1.14645 13.1464C0.95119 13.3417 0.951187 13.6583 1.14645 13.8535C1.34171 14.0488 1.65829 14.0488 1.85355 13.8536L3.85358 11.8536ZM5.35356 13.3535C5.54882 13.1583 5.54882 12.8417 5.35356 12.6464C5.1583 12.4512 4.84171 12.4512 4.64645 12.6464L3.64645 13.6464C3.45119 13.8417 3.45119 14.1583 3.64645 14.3535C3.84171 14.5488 4.1583 14.5488 4.35356 14.3535L5.35356 13.3535ZM9.49997 6.74881C10.1897 6.74881 10.7488 6.1897 10.7488 5.5C10.7488 4.8103 10.1897 4.25118 9.49997 4.25118C8.81026 4.25118 8.25115 4.8103 8.25115 5.5C8.25115 6.1897 8.81026 6.74881 9.49997 6.74881Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Task
              </span>
            </a>
          </div>
        </li>
        <li
          class="space-y-2 p-4"
        >
          <div
            class="flex items-center justify-between gap-2"
          >
            <span
              class="inline-flex px-2 py-0.5 text-xs font-semibold leading-5 rounded items-center bg-badge-3 text-text-expected"
            >
              <svg
                class="h-4 w-4 text-text-expected mr-1"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.99992 1.33301C4.32392 1.33301 1.33325 4.32301 1.33325 7.99967C1.33325 8.33967 1.35925 8.68034 1.40992 9.01234C1.44992 9.27701 1.69592 9.46167 1.96192 9.41768C2.22659 9.37701 2.40792 9.12967 2.36792 8.86567C2.32459 8.58167 2.30259 8.29167 2.30259 7.99967C2.30259 4.85834 4.85859 2.30234 7.99992 2.30234C11.1413 2.30234 13.6973 4.85834 13.6973 7.99967C13.6973 11.1417 11.1413 13.697 7.99992 13.697C6.21792 13.697 4.57192 12.8863 3.48325 11.473C3.31925 11.2617 3.01525 11.221 2.80392 11.385C2.59125 11.5477 2.55192 11.8523 2.71525 12.065C3.98859 13.7177 5.91525 14.6663 7.99992 14.6663C11.6759 14.6663 14.6666 11.6757 14.6666 7.99967C14.6666 4.32301 11.6759 1.33301 7.99992 1.33301Z"
                  fill="currentColor"
                />
                <path
                  d="M10.2265 6.06447L7.41587 8.8758L6.18721 7.64647C5.99187 7.45114 5.67521 7.45114 5.47987 7.64647C5.28454 7.8418 5.28454 8.15847 5.47987 8.3538L7.06187 9.93647C7.15587 10.0298 7.28254 10.0825 7.41587 10.0825C7.54787 10.0825 7.67521 10.0298 7.76921 9.93647L10.9339 6.7718C11.1292 6.57647 11.1292 6.2598 10.9339 6.06447C10.7385 5.86914 10.4219 5.86914 10.2265 6.06447Z"
                  fill="currentColor"
                />
              </svg>
              STARTED
            </span>
            <span
              class="text-xs text-gray-500"
            >
              03:00 PM
            </span>
          </div>
          <a
            class="rounded-md bg-primary-wash p-2 text-sm font-mono block hover:underline"
            href="https://www.youtube.com/watch?v=3"
            rel="noreferrer"
            target="_blank"
          >
            https://www.youtube.com/watch?v=3
          </a>
          <div
            class="flex gap-2"
          >
            <button
              class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none border border-border-primary text-text-primary hover:bg-gray-50 disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 py-[7px] text-[0.875rem] leading-[1.5rem] rounded-md flex-1"
            >
              <svg
                class="mr-1.5"
                fill="none"
                height="20"
                viewBox="0 0 28 28"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m23.674 8.816-5.891-6.198a.867.867 0 0 0-.63-.27h-.24c-.023-.003-.042-.015-.066-.015s-.044.012-.067.014H9.415c-2.89 0-5.331 2.41-5.331 5.261v4.852c0 .487.39.881.872.881a.877.877 0 0 0 .872-.88V7.607c0-1.897 1.642-3.499 3.587-3.499h6.56v2.523c0 2.15 1.73 3.9 3.858 3.906h.001a.877.877 0 0 0 .873-.879.876.876 0 0 0-.87-.883c-1.168-.003-2.117-.965-2.117-2.144V5.097l4.452 4.683v10.413c0 1.977-1.619 3.711-3.465 3.711H9.415c-2.01 0-3.587-1.63-3.587-3.711v-2.944a.877.877 0 0 0-.872-.88.877.877 0 0 0-.873.88v2.944c0 3.069 2.343 5.473 5.332 5.473h9.292c2.824 0 5.21-2.506 5.21-5.473V9.426a.887.887 0 0 0-.243-.61Z"
                  fill="currentColor"
                />
                <path
                  d="M10.307 13.492h7.395a.74.74 0 0 0 .733-.746.74.74 0 0 0-.733-.746h-7.404a.74.74 0 0 0-.733.746.74.74 0 0 0 .733.746h.01ZM10.046 17.345h4.796c.263 0 .476-.327.476-.73 0-.403-.213-.73-.476-.73H10.04c-.262 0-.475.327-.475.73 0 .403.213.73.476.73h.005Z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Log
              </span>
            </button>
            <a
              class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none border border-border-primary text-text-primary hover:bg-gray-50 disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 py-[7px] text-[0.875rem] leading-[1.5rem] rounded-md flex-1"
              href="http://localhost:3000/flower/task/3"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="size-4 mr-1.5"
                fill="none"
                height="15"
                viewBox="0 0 15 15"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.85357 3.85355L7.65355 3.05353C8.2981 2.40901 9.42858 1.96172 10.552 1.80125C11.1056 1.72217 11.6291 1.71725 12.0564 1.78124C12.4987 1.84748 12.7698 1.97696 12.8965 2.10357C13.0231 2.23018 13.1526 2.50125 13.2188 2.94357C13.2828 3.37086 13.2779 3.89439 13.1988 4.44801C13.0383 5.57139 12.591 6.70188 11.9464 7.34645L7.49999 11.7929L6.35354 10.6465C6.15827 10.4512 5.84169 10.4512 5.64643 10.6465C5.45117 10.8417 5.45117 11.1583 5.64643 11.3536L7.14644 12.8536C7.34171 13.0488 7.65829 13.0488 7.85355 12.8536L8.40073 12.3064L9.57124 14.2572C9.65046 14.3893 9.78608 14.4774 9.9389 14.4963C10.0917 14.5151 10.2447 14.4624 10.3535 14.3536L12.3535 12.3536C12.4648 12.2423 12.5172 12.0851 12.495 11.9293L12.0303 8.67679L12.6536 8.05355C13.509 7.19808 14.0117 5.82855 14.1887 4.58943C14.2784 3.9618 14.2891 3.33847 14.2078 2.79546C14.1287 2.26748 13.9519 1.74482 13.6035 1.39645C13.2552 1.04809 12.7325 0.871332 12.2045 0.792264C11.6615 0.710945 11.0382 0.721644 10.4105 0.8113C9.17143 0.988306 7.80189 1.491 6.94644 2.34642L6.32322 2.96968L3.07071 2.50504C2.91492 2.48278 2.75773 2.53517 2.64645 2.64646L0.646451 4.64645C0.537579 4.75533 0.484938 4.90829 0.50375 5.0611C0.522563 5.21391 0.61073 5.34954 0.742757 5.42876L2.69364 6.59928L2.14646 7.14645C2.0527 7.24022 2.00002 7.3674 2.00002 7.50001C2.00002 7.63261 2.0527 7.75979 2.14646 7.85356L3.64647 9.35356C3.84173 9.54883 4.15831 9.54883 4.35357 9.35356C4.54884 9.1583 4.54884 8.84172 4.35357 8.64646L3.20712 7.50001L3.85357 6.85356L6.85357 3.85355ZM10.0993 13.1936L9.12959 11.5775L11.1464 9.56067L11.4697 11.8232L10.0993 13.1936ZM3.42251 5.87041L5.43935 3.85356L3.17678 3.53034L1.80638 4.90074L3.42251 5.87041ZM2.35356 10.3535C2.54882 10.1583 2.54882 9.8417 2.35356 9.64644C2.1583 9.45118 1.84171 9.45118 1.64645 9.64644L0.646451 10.6464C0.451188 10.8417 0.451188 11.1583 0.646451 11.3535C0.841713 11.5488 1.1583 11.5488 1.35356 11.3535L2.35356 10.3535ZM3.85358 11.8536C4.04884 11.6583 4.04885 11.3417 3.85359 11.1465C3.65833 10.9512 3.34175 10.9512 3.14648 11.1465L1.14645 13.1464C0.95119 13.3417 0.951187 13.6583 1.14645 13.8535C1.34171 14.0488 1.65829 14.0488 1.85355 13.8536L3.85358 11.8536ZM5.35356 13.3535C5.54882 13.1583 5.54882 12.8417 5.35356 12.6464C5.1583 12.4512 4.84171 12.4512 4.64645 12.6464L3.64645 13.6464C3.45119 13.8417 3.45119 14.1583 3.64645 14.3535C3.84171 14.5488 4.1583 14.5488 4.35356 14.3535L5.35356 13.3535ZM9.49997 6.74881C10.1897 6.74881 10.7488 6.1897 10.7488 5.5C10.7488 4.8103 10.1897 4.25118 9.49997 4.25118C8.81026 4.25118 8.25115 4.8103 8.25115 5.5C8.25115 6.1897 8.81026 6.74881 9.49997 6.74881Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Task
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
